packages/cdktf/lib/backends/cloud-backend.ts

```diff
@@ -0,0 +1,136 @@
+ */
+export class CloudBackend extends TerraformBackend {
+  constructor(scope: Construct, private readonly props: CloudBackendProps) {
+    super(scope, "backend", "cloud");
+
+    this.node.addValidation(
+      new ValidateBinaryVersion(
+        "terraform",
+        ">=1.1",
+        `${terraformBinaryName} version`,
+        `The cloud block is only supported for Terraform >=1.1. Please upgrade your Terraform version.`
+      )
+    );
+  }
+
+  /**
+   * Adds this resource to the terraform JSON output.
+   */
+  public toTerraform(): any {
+    return {
+      terraform: {
+        cloud: deepMerge(this.synthesizeAttributes(), this.rawOverrides),
+      },
+    };
+  }
+
+  protected synthesizeAttributes(): { [name: string]: any } {
+    return keysToSnakeCase({
+      ...this.props,
+      workspaces: this.props.workspaces.toTerraform(),
+    });
+  }
+
+  public getRemoteStateDataSource(
+    scope: Construct,
+    name: string,
+    _fromStack: string
+  ): TerraformRemoteState {
+    // The cloud "backend" does not have a coresponding remote state data source since it's meant to also
+    // work with multiple workspaces through tags.
+    if ("name" in this.props.workspaces) {
+      return new DataTerraformRemoteState(scope, name, {
+        organization: this.props.organization,
+        workspaces: { name: (this.props.workspaces as any).name },
+        token: this.props.token,
+        hostname: this.props.hostname,
+      });
+    }
+
+    throw new Error(
+      "The Cloud backend only supports cross-stack references when the workspace is defined by name instead of by tags."
+    );
+  }
+}
+
+/**
+ * The Cloud Backend synthesizes a {@link https://www.terraform.io/cli/cloud/settings#the-cloud-block cloud block}.
+ * The cloud block is a nested block within the top-level terraform settings block.
+ * It specifies which Terraform Cloud workspaces to use for the current working directory.
+ * The cloud block only affects Terraform CLI's behavior.
+ * When Terraform Cloud uses a configuration that contains a cloud block - for example, when a workspace is configured to use a VCS provider directly - it ignores the block and behaves according to its own workspace settings.
+ *
+ * https://www.terraform.io/cli/cloud/settings#arguments
+ */
+export interface CloudBackendProps {
+  /**
+   * The name of the organization containing the workspace(s) the current configuration should use.
+   */
+  readonly organization: string;
+
+  /**
+   * A nested block that specifies which remote Terraform Cloud workspaces to use for the current configuration.
+   * The workspaces block must contain exactly one of the following arguments, each denoting a strategy for how workspaces should be mapped:
+   */
+  readonly workspaces: NamedCloudWorkspace | TaggedCloudWorkspaces;
```

Comment:
Does this work via JSII? Or does this result in e.g. `interface{}` or `any`?
If yes, it might make sense to combine them into a single class that validates that either `name` or `tags` is set.

Reply:
Yes it works with JSII, we do it the same way for remote